src/mono/wasm/debugger/BrowserDebugProxy/ValueTypeClass.cs

```diff
@@ -83,8 +83,24 @@ public ValueTypeClass(byte[] buffer, string className, JArray fields, int typeId
                     && !f.Attributes.HasFlag(FieldAttributes.Static));
             foreach (var field in writableFields)
             {
-                lastWritableFieldValue = await sdbAgent.ValueCreator.ReadAsVariableValue(cmdReader, field.Name, token, isOwn: true, field.TypeId, forDebuggerDisplayAttribute: false);
-                fields.Add(GetFieldWithMetadata(field, lastWritableFieldValue, isStatic: false));
+                //check if it's fixed size array and behave as a inline array
+                ElementType etype = (ElementType)cmdReader.ReadByte();
+                if (etype == (ElementType)ValueTypeId.FixedArray && writableFields.Count() == 1)
+                {
+                    ElementType elementType = (ElementType)cmdReader.ReadByte();
+                    var arraySize = cmdReader.ReadInt32();
+                    inlineArray = new(arraySize + 1);
+                    for (int i = 0; i < arraySize; i++)
+                    {
+                        inlineArray.Add(await sdbAgent.ValueCreator.CreateFixedArrayElement(cmdReader, elementType, $"{i}", token));
+                    }
```

Comment:
nit: this could probably be a method that takes in the type details, and the array size to return `inlineArray`, avoiding the multiple the awaits too.